soroban-sdk-auth/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![no_std]
+
+use soroban_sdk::{serde::Serialize, Account, BigInt, BytesN, Env, EnvVal, Symbol, TryIntoVal};
+
+pub mod public_types;
+use crate::public_types::{
+    Identifier, KeyedAccountAuthorization, KeyedAuthorization, KeyedEd25519Signature, Message,
+    MessageV0,
+};
+
+pub trait NonceAuth {
+    fn read_nonce(e: &Env, id: Identifier) -> BigInt;
+    fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt;
+    fn get_keyed_auth(&self) -> &KeyedAuthorization;
+}
+
+pub fn check_ed25519_auth(
+    e: &Env,
+    auth: &KeyedEd25519Signature,
+    function: Symbol,
+    parameters: EnvVal,
+) {
+    let msg = MessageV0 {
+        function,
+        contrct_id: e.get_current_contract(),
+        network_id: e.ledger().network_passphrase(),
+        parameters: parameters.to_raw().try_into_val(e).unwrap(),
+    };
+    let msg_bin = Message::V0(msg).serialize(e);
+
+    e.verify_sig_ed25519(
+        auth.public_key.clone().into(),
+        msg_bin,
+        auth.signature.clone().into(),
+    );
+}
+
+pub fn check_account_auth(
+    e: &Env,
+    auth: &KeyedAccountAuthorization,
+    function: Symbol,
+    parameters: EnvVal,
+) {
+    let acc = Account::from_public_key(&auth.public_key).unwrap();
+
+    let msg = MessageV0 {
+        function,
+        contrct_id: e.get_current_contract(),
+        network_id: e.ledger().network_passphrase(),
+        parameters: parameters.to_raw().try_into_val(e).unwrap(),
+    };
+    let msg_bin = Message::V0(msg).serialize(e);
```

Comment:
Let's use the term `bytes` instead of `bin` because in the SDK the type we store bytes in is `Bytes`.
```suggestion
    let msg_bytes = Message::V0(msg).serialize(e);
```